rust/src/producers/workspace.rs

```diff
@@ -37,15 +44,10 @@ impl Sink for WorkspaceSink {
     }
 
     fn push_constraints(&mut self, cs: ConstraintSystem) -> Result<()> {
-        let file = match self.constraints_file {
-            None => {
-                self.constraints_file = Some(File::create(
-                    self.workspace.join("constraints.zkif"))?);
-                self.constraints_file.as_mut().unwrap()
-            }
-            Some(ref mut file) => file,
-        };
-
+        let mut constraints_file = Some(File::create(
+                    self.workspace.join(format!("constraints_{}.zkif", &self.cs_file_counter)))?);
+        self.cs_file_counter += 1;
+        let file = constraints_file.as_mut().unwrap();
```

Comment:
Just remove the `Some` above then this line won't be necessary.

Reply:
done in new commit